pages/profile/x/[userName].tsx

```diff
@@ -0,0 +1,38 @@
+import { useQuery } from "@apollo/client";
+import { useRouter } from "next/router";
+import React, { useState } from "react";
+import Header404 from "../../../components/Header404";
+import { FETCH_USER } from "../../../graphql/fetchUser";
+import InternalProfile from "../v2/[userId]";
+
+export default function ProfileGateway() {
+  const router = useRouter();
+  const { userName } = router.query;
+  const [validUserId, setValidUserId] = useState<string>();
+
+  const { loading } = useQuery<{ users: Array<{ id: string }> }>(FETCH_USER, {
+    variables: {
+      link: userName,
+    },
+    onCompleted: (data) => {
+      if (data.users.length > 0) {
+        setValidUserId(data.users[0].id);
+      }
+    },
+  });
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  if (validUserId) {
```

Comment:
I think this check should be done server side instead of client side. You can move this check to server side by making use of getServerSideProps

https://reacthustle.com/blog/redirect-to-404-nextjs-dynamic-routes
https://nextjs.org/docs/api-reference/data-fetching/get-server-side-props

The react hustle article explains why that approach is better.